feat: Add Persian Red Dark Mode Theme

0xken opened the pull request:
New dark theme with red accents

MatthewJohn commented:
Hey @0xken,

Thank you for this :)

I've got a couple of questions/requests, if I may:

Some of the non-text elements (such as headings to filters and the body of 'cards' are quite bright:
![Screenshot from 2023-06-08 06-47-21](https://github.com/MatthewJohn/terrareg/assets/1266262/e48a41a3-b72c-41d4-8231-266aa24626a6)
![Screenshot from 2023-06-08 06-46-47](https://github.com/MatthewJohn/terrareg/assets/1266262/672fa477-9902-47a4-a48b-0eda9becd996)
I wonder if it would be worth making them slightly less bright, as, to me, they seem to stand out quite heavily.
There also seem to be some white gaps around borders with radiuses (or radii), I wonder if this could be addressed at all? :)

One other thing I'd request, if possibly, could the name of the theme be changed to something that has no relation to any ethnicity or culture :)

Thanks
Matt

MatthewJohn commented:
One other question - is there any way this could be minified (it appears to be considerably larger than the pre-existing themes) - is this a theme that you've written yourself - I'm struggling to find other references to it on github? :)

0xken commented:
> Some of the non-text elements (such as headings to filters and the body of 'cards' are quite bright

These colors have been adjust in the updated PR. The screenshots are below showing the updates.

> There also seem to be some white gaps around borders with radiuses (or radii), I wonder if this could be addressed at all? :)

This is resolved now see the screenshot below to confirm that's what you were referencing.

> One other thing I'd request, if possibly, could the name of the theme be changed to something that has no relation to any ethnicity or culture :)

Naming is hard. It was just the name of the color I used, however I changed it to `Cherry Dark`. I would like to make it easier to inject themes at some point so we don't have to update core code or templates, but that's for another time :)

 > is there any way this could be minified (it appears to be considerably larger than the pre-existing themes) - is this a theme that you've written yourself - I'm struggling to find other references to it on github? :)

I have added the compression to the CSS output. This is a theme I crafted using the Bulma framework, which is what the existing themes were using. 

The source of the theme is located at `sass/cherry-dark.scss`. You can make changes then run `npm run css-build` to update the compressed css file the server uses. 


If you have any other questions, complaints, or suggestions let me know and I'll tackle it. 

Thanks,
Ken

![Screenshot_2023-06-09_00-44-02](https://github.com/MatthewJohn/terrareg/assets/62523933/e915a0e9-702d-493c-8d31-f2ad86668fce)

![Screenshot_2023-06-09_00-44-49](https://github.com/MatthewJohn/terrareg/assets/62523933/a670be3b-9a1b-4340-a018-23374bae8e55)

MatthewJohn commented:
Hey Ken,

Apologies for the slow reply - it's been a bit of a hectic weekend!

> These colors have been adjust in the updated PR. The screenshots are below showing the updates.
> ...
> This is resolved now see the screenshot below to confirm that's what you were referencing.

Thank you :)

> Naming is hard. It was just the name of the color I used, however I changed it to Cherry Dark. I would like to make it easier to inject themes at some point so we don't have to update core code or templates, but that's for another time :)

Absolutely :) Cherry Dark sounds great!

> I have added the compression to the CSS output. This is a theme I crafted using the Bulma framework, which is what the existing themes were using.
> The source of the theme is located at sass/cherry-dark.scss. You can make changes then run npm run css-build to update the compressed css file the server uses.

Awesome - As you can see, there hasn't been any effort to use any node-esque frameworks yet. When that time comes, we can deal with adapting the npm commands to work :)

I'll raise an upstream PR now and post a link :)

MatthewJohn commented:
I've created a PR here: https://gitlab.dockstudios.co.uk/pub/terrareg/-/merge_requests/333# :)
